server/graphql/v2/query/collection/ActivitiesCollectionQuery.ts

```diff
@@ -47,36 +53,34 @@ const ActivitiesCollectionQuery = {
   args: ActivitiesCollectionArgs,
   async resolve(_: void, args, req): Promise<CollectionReturnType> {
     const { offset, limit } = args;
-    const account = await fetchAccountWithReference(args.account, { throwIfMissing: true });
+    let accounts;
+    if (args.account) {
+      accounts = await fetchAccountsWithReferences(args.account, { throwIfMissing: true });
+    }
 
     // Check permissions
     checkRemoteUserCanUseAccount(req);
-    if (!req.remoteUser.isAdminOfCollective(account) && !req.remoteUser.isRoot()) {
-      return { nodes: null, totalCount: 0, limit, offset };
-    }
-
-    // Filter activities by account based on attribution. null = all related activities
+    const isRootUser = req.remoteUser.isRoot();
     const accountOrConditions = [];
-    const getDistinctFromCondition = value => ({ [Op.or]: [{ [Op.is]: null }, { [Op.ne]: value }] }); // Ideally, we should use `IS DISTINCT FROM` (https://wiki.postgresql.org/wiki/Is_distinct_from), but it's not supported by Sequelize yet: https://github.com/sequelize/sequelize/issues/12612
+    const where = { [Op.or]: accountOrConditions };
+    for (const account of accounts) {
+      if (isRootUser || req.remoteUser.isAdminOfCollective(account)) {
+        accountOrConditions.push({ CollectiveId: account.id });
+        if (args.includeChildrenAccounts) {
+          const childIds = await account.getChildren().then(children => children.map(child => child.id));
+          accountOrConditions.push(...childIds.map(id => ({ CollectiveId: id })));
+        }
+        if (args.includeHostedAccounts && account.isHostAccount) {
+          const hostedAccounts = await account.getHostedCollectives({ attributes: ['id'] });
+          accountOrConditions.push(...hostedAccounts.map(hostedAccount => ({ CollectiveId: hostedAccount.id })));
+        }
```

Comment:
If you're fetching 30 accounts here, you'll basically trigger 60 additional SQL queries. This is a massive cost that we can easily avoid: 

```suggestion
        accountOrConditions.push({ CollectiveId: account.id });
        if (args.includeChildrenAccounts) {
          accountOrConditions.push({ ParentCollectiveId: account.id });
        }
        if (args.includeHostedAccounts && account.isHostAccount) {
          accountOrConditions.push({ HostCollectiveId: account.id, isActive: true });
        }
```

Reply:
@Betree : For the `if (args.includeHostedAccounts && account.isHostAccount) {` part this makes sense. But do you think we could leave the `if (args.includeChildrenAccounts) {` as is? Note that `Activity` doesn't include the `ParentCollectiveId`.

Reply:
You can join on `CollectiveId` them